src/planner/drop_plan.cpp

```
@@ -25,14 +25,31 @@ DropPlan::DropPlan(const std::string &name) {
 }
 
 DropPlan::DropPlan(parser::DropStatement *parse_tree) {
-  if (parse_tree->type == parser::DropStatement::EntityType::kTable) {
-    table_name = parse_tree->GetTableName();
-    missing = parse_tree->missing;
-  } else if (parse_tree->type == parser::DropStatement::EntityType::kTrigger) {
-    // note parse_tree->table_name is different from parse_tree->GetTableName()
-    table_name = std::string(parse_tree->table_name_of_trigger);
-    trigger_name = std::string(parse_tree->trigger_name);
-    drop_type = DropType::TRIGGER;
+  switch (parse_tree->type) {
+    case parser::DropStatement::EntityType::kDatabase :{
+      database_name = parse_tree->GetDatabaseName();
+      missing = parse_tree->missing;
+      drop_type = DropType::DB;
+      break;
+    }
+    case parser::DropStatement::EntityType::kTable: {
+      database_name = parse_tree->GetDatabaseName();
+      table_name = parse_tree->GetTableName();
+      missing = parse_tree->missing;
+      drop_type = DropType::TABLE;
+      break;
+    }
+    case parser::DropStatement::EntityType::kTrigger: {
+      // note parse_tree->table_name is different from parse_tree->GetTableName()
+      database_name = parse_tree->GetDatabaseName();
+      table_name = std::string(parse_tree->table_name_of_trigger);
+      trigger_name = std::string(parse_tree->trigger_name);
+      drop_type = DropType::TRIGGER;
+      break;
+    }
+    default: {
+      LOG_ERROR("Not supported Drop type");
```

@haojin2 Should we throw an exception here?